File: Eunleelee/2024-02/2월 6일/폰켓몬.py
def solution(nums):
    answer = 0
    a = len(nums) / 2 #가져갈 수 있는 포켓몬의 수
    num = set(nums) # 중복제거
    
    if len(num) < a:
        answer = len(num)
    else:
        answer = a
    
    return answer

File: jja6312/2월 6일/폰켓몬.java
import java.util.*;
class Solution {
        public int solution(int[] nums) {
            int answer = 0;
            Set set = new HashSet();
            for(int num : nums){
                set.add(num);
            }
            int maxVariation = set.size();
            if(maxVariation>nums.length/2){
                answer = nums.length/2;
            }else{
                answer = maxVariation;
            }
                    
            
            return answer;
        }
    }